modules/core/src/model/PurposeRestrictionVector.ts

```diff
@@ -134,7 +134,7 @@ export class PurposeRestrictionVector extends Cloneable<PurposeRestrictionVector
    * @param {number[]|null|undefined} vendorsIds
    * @return {void}
    */
-  public restrictPurposeToLegalBasis(purposeRestriction: PurposeRestriction, vendorsIds: number[] = Array.from(this.gvl.vendorIds)): void {
+  public restrictPurposeToLegalBasis(purposeRestriction: PurposeRestriction, vendorsIds: number[] = [...this.gvl.vendorIds]): void {
 
```

Comment:
Something like
```
        const vendorsIds = [];
        
        this.gvl.vendorIds.forEach((value) =>
            vendorsIds.push(value)
        );
 ```
 could be used. However `Array.from` is expecting an iterable entity, just like the spread operator is.